snooty/specparser.py

```diff
@@ -1,12 +1,15 @@
 """Parser for a TOML spec file containing definitions of all supported reStructuredText
    directives and roles, and what types of data each should expect."""
 
+from __future__ import annotations
```

Comment:
For my edification, what's this doing?

Reply:
Python has a mechanism for adopting small language changes; before changes get turned on by default, you can import from `__future__` to opt into that feature.

In this case, it's upgrading past an annoying quirky interaction of type checking and the language where you can't reference a class within type annotations inside of that same class, because the class definition isn't yet complete. For example, this would be no good:

```
class A:
    def foo(self, other: A):
        pass
```

Instead, you would have to do:


```
class A:
    def foo(self, other: "A"):
        pass
```

This changes in Python 3.8, and the first form becomes valid. But because we're currently targeting Python 3.7, we have to do this import to opt into the Python 3.8 behavior.